lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/library/validation/ValidatorService.java

```diff
@@ -0,0 +1,162 @@
+package uk.gov.di.ipv.cri.kbv.api.library.validation;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.jwt.proc.BadJWTException;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import uk.gov.di.ipv.cri.kbv.api.library.domain.SessionRequest;
+import uk.gov.di.ipv.cri.kbv.api.library.exception.ClientConfigurationException;
+import uk.gov.di.ipv.cri.kbv.api.library.exception.ValidationException;
+import uk.gov.di.ipv.cri.kbv.api.library.service.ConfigurationService;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.security.PublicKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Map;
+
+public class ValidatorService {
```

Comment:
Is there a `ValidatorService.Test` somewhere, I think you should bring the test across from where this was copied

Reply:
Added new test for this service - `ValidatorServiceTest`